### PR TITLE
feat: add `.toEitherT` to `Validated`

### PR DESCRIPTION
**Pull request by jeengbe**
Similar to `.toEither` but with `EitherT`

### CODE REVIEW

File: core/src/main/scala/cats/data/Validated.scala [hunk 1]
```diff
@@ -265,6 +265,30 @@ sealed abstract class Validated[+E, +A] extends Product with Serializable {
       case Valid(a)   => Right(a)
     }
 
+  /**
+   * Converts the value to an EitherT[F, E, A]
+   *
+   * Example:
+   * {{{
+   * scala> import cats.syntax.all._
+   * scala> given ExecutionContext = ExecutionContext.global
```
Comment (jeengbe) [inline core/src/main/scala/cats/data/Validated.scala:274]: I believe this is Scala 3 syntax, please replace it with the Scala 2 equivalent (which I assume doc comments are written in)

**Comment by jeengbe** [comment]
Hm, the errors correctly mark:

```log
[error] /home/runner/work/cats/cats/core/src/main/scala/cats/data/Validated.scala:286:7: covariant type E occurs in invariant position in type [F[_]](implicit evidence$1: cats.Applicative[F]): cats.data.EitherT[F,E,A] of method toEitherT
[error]   def toEitherT[F[_]: Applicative]: EitherT[F, E, A] =
[error]       ^
```

yet with an extension, this variance difference is no issue:

```scala
extension [E, A](validated: Validated[E, A])
  def toEitherT[F[_]: Applicative]: EitherT[F, E, A] = validated match
    case Validated.Invalid(e) => EitherT.leftT(e)
    case Validated.Valid(a)   => EitherT.rightT(a)
```

I have unfortunately not the slightest idea how to fix this

File: core/src/main/scala/cats/data/Validated.scala [hunk 2]
```diff
@@ -265,6 +265,30 @@ sealed abstract class Validated[+E, +A] extends Product with Serializable {
...
+   *
+   * scala> v2.toEitherT[Future].value
+   * res1: Future[Either[String, Int]] = Future(Right(123))
+   * }}}
+   */
+  def toEitherT[F[_]: Applicative]: EitherT[F, E, A] =
+    this match {
+      case Invalid(e) => EitherT.leftT(e)
+      case Valid(a)   => EitherT.rightT(a)
+    }
```
Comment (jeengbe) [inline core/src/main/scala/cats/data/Validated.scala:290]: And perhaps

```suggestion
    EitherT.fromEither(this.toEither)
```

is cleaner

**Review by rossabaker — COMMENTED** [review]
Thanks for the contribution!  The compilation error can be fixed with:

```
  def toEitherT[F[_]: Applicative, EE >: E, AA >: A]: EitherT[F, EE, AA] =
```

... but that's not what we want, because then `.toEitherT` needs three parameters, and it loses its convenience.  I thought `Ior.toIorT` might serve as prior art, but it doesn't exist.  But `IorT.fromIor` does!  Maybe what we want is `EitherT.fromValidated`?

**Comment by satorg** [comment]
> Maybe what we want is `EitherT.fromValidated`?

Agree. Adding a new constructor to `EitherT` seems more coherent taking into account that it already offers constructors like `fromEither`, `fromOption` and some others.

Another thought is that `EitherT` is more complex type comparing to `Validated`, so I personally feel that making more complex abstractions depending on more basic ones is usually preferable that doing the opposite.

**Comment by jeengbe** [comment]
Thanks for outlining the reasoning, I agree.

**Pull request closed without merging**